src/main/java/com/google/devtools/build/lib/bazel/repository/skylark/SkylarkRepositoryContext.java

```diff
@@ -192,30 +192,36 @@ public void symlink(Object from, Object to, Location location)
     }
   }
 
-  private void checkInOutputDirectory(SkylarkPath path) throws RepositoryFunctionException {
+  private void checkInOutputDirectory(String operation, SkylarkPath path) throws RepositoryFunctionException {
```

Comment:
This really should be a bool or enum instead of a string. If you use a bool, the call sites should specify the param name in a comment, as in

```
checkInOutputDirectory(/*isRead=*/ true, ...)
```

Reply:
I think a `bool` would be inappropriate here, because there's no reason it wouldn't use more specific error messages for other checks in the future. I only copy-pasted `"write"` into existing callers to avoid significant diffs in unrelated tests.

Reply:
I wasn't suggesting changing the error message for the existing case, I just wanted to avoid passing strings if it's only ever allowed to take on two possible values (e.g. to avoid subtle misspellings leading to errors). That said I see that this is only used for the error message, not to control the actual operation being done, so it's fine to leave it a string.